### PR TITLE
WIP: Admin action to recreate a tournament

### PR DESCRIPTION
**Pull request by n8kim1**
Blocked on #723 for safety

The code quality -- especially computation of new name -- is a bit scuffed but it does the job for now.
I don't think this needs clean error handling yet, so I'd rather add it later as needed, but happy to do so now.

Does part of #594

### CODE REVIEW

**Comment by j-mao** [comment]
Is there a reason this is urgent? If not, would rather wait for cleaner code before considering this for merge. In particular am curious whether it's possible to avoid hardcoded naming patterns (which have unintended consequences, such as failing "bc24-very-cool-tournament").

**Comment by n8kim1** [comment]
Just don't write silly names

**Comment by n8kim1** [comment]
but also, mostly cuz if sprint tour fails, then this PR would be great for safety in copying it over, as well as time/convenience

File: backend/siarnaq/api/episodes/models.py [hunk 1]
```diff
@@ -305,6 +305,50 @@ def initialize(self):
...
+        # Generally append -v2 or increment this number if already present
+        name_split = self.name_short.split("-v")
+        if len(name_split) == 1:
+            # -v not found, add it
+            name_short_new = self.name_short + "-v2"
+        else:
+            # -v found. Split the name parts, and increment to the number
+            # This might be a little weird if multiple are found
+            name_prefix, version_tag = name_split
+            version_tag_number_new = int(version_tag) + 1
```
[the remark this answers is not in the document]
Comment (j-mao) [reply backend/siarnaq/api/episodes/models.py:325]: quite literally "bc24-very-cool-tournament"

[the remark this answers is not in the document]
Comment (n8kim1) [reply backend/siarnaq/api/episodes/models.py:325]: my

**Comment by n8kim1** [comment]
declining this PR --

abandoning a tournament, in the way this PR does, is problematic
> thoughts on prioritizing the other things first- main objection to dupe tournament is that it mainstreams “give up on the tournament whenever anything goes wrong and leave junk behind in db”
the fix feels more like a “dirty fix” which was ok last year considering we were still in alpha, but feels less ok for something we want to enshrine in stable prod
(for a full discussion: i think it’s this combination of give-up + junk that i object to, not give-up alone. for example if there was no junk left behind, then i wouldn’t feel so irked)

and solving the problem posed here can be done in other ways. for example, consider this outline

> rescind match result on challonge
prevent requeueing a tournament match that’s already in db (regardless of its state: even if it’s errored, it should be re-queued or deleted+recreated)
change challonge participant list (in case seeds or eligibility was wrong) (should pre-condition complete rescind before allowing this)

**Pull request closed without merging**